assets/src/hooks/usePersistedStateReducer.ts

```diff
@@ -64,17 +68,30 @@ const usePersistedStateReducer = (): [State, Dispatch] => {
   }, [selectedRouteIds, ladderDirections, ladderCrowdingToggles])
 
   useEffect(() => {
-    if (routeTabsToPush && !routeTabsPushInProgress) {
+    if (routeTabsToPush[0] && !routeTabsPushInProgress) {
       dispatch(startingRouteTabsPush())
-      putRouteTabs(routeTabsToPush)
+      putRouteTabs(routeTabsToPush[0])
         .then((response) => {
           if (response.ok) {
+            setRouteTabsPushRetriesLeft(routeTabsPushRetries)
+            dispatch(routeTabsPushComplete())
+          } else if (routeTabsPushRetriesLeft > 0) {
+            setRouteTabsPushRetriesLeft(routeTabsPushRetriesLeft - 1)
+            dispatch(retryRouteTabsPushIfNotOutdated(routeTabsToPush[0]))
+          } else {
+            setRouteTabsPushRetriesLeft(routeTabsPushRetries)
             dispatch(routeTabsPushComplete())
+          }
+        })
+        .catch(() => {
```

Comment:
It looks like, per MDN's [fetch API docs](https://developer.mozilla.org/en-US/docs/Web/API/Response/ok), `response.ok` will only be truthy if the response code is in the 2xx range.

Reply:
Here is the full list of reasons that may cause `fetch` to throw an error and end up in `catch` rather than resolving with `response.ok` being false: https://developer.mozilla.org/en-US/docs/Web/API/fetch#exceptions

The main one we would have to worry about is the very last one (network error).